src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -855,6 +859,211 @@ class TextimageOnOptionSegment(BaseSegment):
+    """
+
+    _ledger_view_option = Delimited(
+        Sequence(
+            "TRANSACTION_ID_COLUMN_NAME",
+            Ref("EqualsSegment"),
+            Ref("ColumnReferenceSegment"),
+            optional=True,
+        ),
+        Sequence(
+            "SEQUENCE_NUMBER_COLUMN_NAME",
+            Ref("EqualsSegment"),
+            Ref("ColumnReferenceSegment"),
+            optional=True,
+        ),
+        Sequence(
+            "OPERATION_TYPE_COLUMN_NAME",
+            Ref("EqualsSegment"),
+            Ref("ColumnReferenceSegment"),
+            optional=True,
+        ),
+        Sequence(
+            "OPERATION_TYPE_DESC_COLUMN_NAME",
+            Ref("EqualsSegment"),
+            Ref("ColumnReferenceSegment"),
+            optional=True,
+        ),
+    )
+
+    type = "table_option_statement"
+
+    match_grammar = Sequence(
+        "WITH",
+        Bracketed(
+            Delimited(
+                AnyNumberOf(
+                    Sequence("MEMORY_OPTIMIZED", Ref("EqualsSegment"), "ON"),
+                    Sequence(
+                        "DURABILITY",
+                        Ref("EqualsSegment"),
+                        OneOf("SCHEMA_ONLY", "SCHEMA_AND_DATA"),
+                    ),
+                    Sequence(
+                        "SYSTEM_VERSIONING",
+                        Ref("EqualsSegment"),
+                        "ON",
+                        Sequence(
+                            Bracketed(
+                                "HISTORY_TABLE",
+                                Ref("EqualsSegment"),
+                                Ref("TableReferenceSegment"),
+                                Sequence(
+                                    Ref("CommaSegment"),
+                                    "DATA_CONSISTENCY_CHECK",
+                                    Ref("EqualsSegment"),
+                                    OneOf("ON", "OFF"),
+                                    optional=True,
+                                ),
+                            ),
+                            optional=True,
+                        ),
+                    ),
+                    Sequence(
+                        "DATA_COMPRESSION",
+                        OneOf(
+                            "NONE",
+                            "ROW",
+                            "PAGE",
+                        ),
+                        Sequence(
+                            "ON PARTITONS",
+                            Bracketed(
+                                Delimited(
+                                    Ref("NumericLiteralSegment"),
+                                ),
+                            ),
+                            optional=True,
+                        ),
+                    ),
+                    Sequence(
+                        "XML_COMPRESSION",
+                        OneOf("ON", "OFF"),
+                        Sequence(
+                            "ON PARTITONS",
```

Comment:
Typo in word partitions here, also, we don't include spaces here, so can you change it to 
```
"ON",
"PARTITIONS"
```